test-utils/junit5-extension/src/test/java/org/operaton/bpm/engine/test/junit5/ProcessEngineExtensionServiceInjectionTest.java

```diff
@@ -0,0 +1,51 @@
+package org.operaton.bpm.engine.test.junit5;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.operaton.bpm.engine.*;
+import org.operaton.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(ProcessEngineExtension.class)
+class ProcessEngineExtensionServiceInjectionTest {
+    ProcessEngine processEngine;
+    ProcessEngine processEngine2;
```

Comment:
Maybe add a comment why there is `processEngine` and `processEngine2`? I guess you are testing that a class which was referred twice has the same reference, but I stumbled over this and had to check / think for a short time about it

Reply:
Yes, the original logic was to set any field of that type. I think this is a corner case that should not be used. Maybe it is even better to enforce that only one field of that kind exists?
I have added the comment.